addons/purchase_stock/models/account_move_line.py

```diff
@@ -353,7 +353,8 @@ def _prepare_pdiff_aml_vals(self, qty, unit_valuation_difference):
                 'balance': self.currency_id._convert(
                     qty * price * sign,
                     self.company_currency_id,
-                    self.company_id, fields.Date.today(),
+                    self.company_id,
+                    self.move_id.invoice_date or self.date,
                 ),
```

Comment:
Isn't it possible to use `currency_rate` ?
https://github.com/odoo/odoo/blob/bb08a627528cd7402a16689c420f3013488e9a4d/addons/account/models/account_move_line.py#L115-L118